src/MySqlConnector/MySqlClient/MySqlBulkLoader.cs

```diff
@@ -153,15 +153,22 @@ private string BuildSqlCommand()
             if (!string.IsNullOrWhiteSpace(FileName) && SourceStream != null)
                 throw new InvalidOperationException("Cannot set both FileName and SourceStream");
 
-            if (string.IsNullOrWhiteSpace(FileName) && SourceStream != null)
-            {
-                if (!Local)
-                    throw new InvalidOperationException("Cannot use SourceStream when Local is not true.");
+			// LOCAL INFILE case
+			if (Local && !string.IsNullOrWhiteSpace(FileName) && SourceStream == null)
```

Comment:
We know that `SourceStream == null` due to the test on line 153; it's redundant to check it again here.